lambdas/farsetlabs/handler.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+
+const icalendar = require('icalendar');
+const {
+  createHash,
+  getDateTimePathFor,
+  getFromWeb,
+  setInS3
+} = require("aws-lambda-data-utils");
+const { bucketName, getEventsUrl } = require("./config");
+
+const getFromApi = async function() {
+  const initialResponse = await getFromWeb(getEventsUrl());
+  const calendar = icalendar.parse_calendar(initialResponse);
+
+  if (calendar.events().length < 1) {
+    throw new Error("No events found");
+  }
+
+  return initialResponse;
+}
+
+const uploadTo = function(createFilename, fileContents) {
+  const today = new Date();
+  const hash = createHash(fileContents);
+  const prefix = getDateTimePathFor(today);
+  const filename = createFilename(today, hash);
+  const filePath = `${prefix}/${filename}`;
+
+  return setInS3(prefix, bucketName, filePath, fileContents);
+};
+
+const uploadData = function(calendarData) {
+    return uploadTo(
+      (today, hash) =>
+        `farset-labs-calendar__${today.valueOf()}__${hash}.ics`,
+        calendarData
+    );
+};
+
+module.exports.hello = async (event, context, callback) => {
```

Comment:
It might be nice to replace the default function of `hello` with `produce`. This way when we have multiple lambdas for each source, we can break it down to `produce`, `consume`, `transform`, etc. If this changes, the serverless.yml will also need updated

Reply:
@apoclyps Good idea. Will punt for now to keep consistency, and tackle as part of the next stage of writing the transform logic for each source